scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/ScheduleFailure.java

```diff
@@ -0,0 +1,45 @@
+package gov.nasa.jpl.aerie.scheduler.server.services;
+
+import javax.json.JsonValue;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public record ScheduleFailure(
+    String type,
+    String message,
+    JsonValue data,
+    String trace
+) {
+  public static final class Builder {
+    private String type = "";
+    private String message = "";
+    private String trace = "";
+    private JsonValue context = JsonValue.EMPTY_JSON_OBJECT;
+
+    public Builder type(final String type) {
+      this.type = type;
+      return this;
+    }
+
+    public Builder message(final String message) {
+      this.message = message;
+      return this;
+    }
+
+    public Builder trace(final Throwable throwable) {
+      final var sw = new StringWriter();
+      throwable.printStackTrace(new PrintWriter(sw));
+      this.trace = sw.toString();
+      return this;
+    }
+
+    public Builder data(final JsonValue context) {
```

Comment:
🔵 Curious about the choice to name the method `data` but the field `context` - any reason for them to be different?